### PR TITLE
Refactor 'ForwardsTransactionIds' trait.

### PR DESCRIPTION
**Pull request by DFurnes**
### What's this PR do?
This pull request adds documentation & a test case for the `ForwardsTransactionIds` trait. All the logic for generating, incrementing, and logging the request IDs is internal to the trait so it doesn't need to end up littering application code.

### How should this be reviewed?
I've written a test case to demonstrate the intended behavior, and tested this out with my local Rogue instance (confirming that Blink API requests are logged just like before).

### Checklist
- [x] Tests added for new features/bug fixes.
- [ ] Is this a [breaking change](http://semver.org)?

### CODE REVIEW

File: src/ForwardsTransactionIds.php [hunk 1]
```diff
@@ -6,44 +6,33 @@
...
+            $newHeader = ['X-Request-ID' => uniqid() . '-0'];
         } else {
-            // Else, if there is a 'X-Request-ID' in the header, get transaction ID and increment the step at the end of Transaction ID.
-            $transactionIdParts = explode('-', $transactionId);
-            $transactionIdBase = $transactionIdParts[0];
-            $incrementedStep = $transactionIdParts[1] + 1;
-            $transactionIdHeader = ['X-Request-ID' => $transactionIdBase . '-' . $incrementedStep];
+            // Otherwise, if there is a 'X-Request-ID' header, keep the
+            // current transaction ID and increment the step by one.
+            list($requestId, $step) = explode('-', $transactionId, 2);
```
Comment (chloealee) [inline src/ForwardsTransactionIds.php:22]: never knew about `list` before - so cool!

**Review by sbsmith86 — APPROVED** [review]
It looks great to me! I think we should make a ticket to clean up the rogue transaction ID logic.

**Comment by chloealee** [comment]
This looks great to me! Are there places in other apps that we need to delete this logic from (Rogue?) at the same time as merging this?

**Review by chloealee — APPROVED** [review]

**Comment by chloealee** [comment]
ah same thought @sbsmith86 ! :)

**Comment by DFurnes** [comment]
Yeah, agreed! I can see about tidying those up alongside a PR to pull this into Rogue. Luckily, no other apps implemented their own Transaction ID logic so that should be it for cleanup!

**Pull request merged**